packages/flutter/lib/src/painting/linear_border.dart

```diff
@@ -0,0 +1,374 @@
+/// start and end depend on the current [TextDirection], see
+/// [Directionality].
+@immutable
+class LinearBorderEdge {
+  /// Defines one side of a [LinearBorder].
+  ///
+  /// The values of [size] and [alignment] must be between
+  /// 0 and 1, and -1 and 1 respectively.
+  const LinearBorderEdge({
+    this.size = 1,
+    this.alignment = 0,
+  }) : assert(size >= 0 && size <= 1);
+
+  /// A value between 0 and 1 that defines the length of the edge as a
+  /// percentage of the length of the corresponding box
+  /// edge. Default is 1.
+  final double size;
+
+
+  /// A value between -1 and 1 that defines how edges for which [size]
+  /// is less than 1 are aligned relative to the corresponding box edge.
+  ///
+  ///  * -1, aligned in the "start" direction. That's left
+  ///    for [TextDirection.ltr] and right for [TextDirection.rtl].
+  ///  * 0, centered.
+  ///  * 1, aligned in the "end" direction. That's right
+  ///    for [TextDirection.ltr] and left for [TextDirection.rtl].
+  final double alignment;
+
+  /// Linearly interpolates between two [LinearBorder]s.
+  ///
+  /// If both `a` and `b` are null then null is returned. If `a` is null
+  /// then we interpolate to `b` varying [size] from 0 to `b.size`. If `b`
+  /// is null then we interpolate from `a` varying size from `a.size` to zero.
+  /// Otherwise both values are interpolated.
+  static LinearBorderEdge? lerp(LinearBorderEdge? a, LinearBorderEdge? b, double t) {
+    if (a == null && b == null) {
+      return null;
+    }
+
+    a ??= LinearBorderEdge(alignment: b!.alignment, size: 0);
+    b ??= LinearBorderEdge(alignment: a.alignment, size: 0);
+
+    return LinearBorderEdge(
+      size: lerpDouble(a.size, b.size, t)!,
+      alignment: lerpDouble(a.alignment, b.alignment, t)!,
+    );
+  }
+
+  @override
+  bool operator ==(Object other) {
+    if (identical(this, other)) {
+      return true;
+    }
+    if (other.runtimeType != runtimeType) {
+      return false;
+    }
+    return other is LinearBorderEdge
+        && other.size == size
+        && other.alignment == alignment;
+  }
+
+  @override
+  int get hashCode => Object.hash(size, alignment);
+
+  @override
+  String toString() {
+    final StringBuffer s = StringBuffer('${objectRuntimeType(this, 'LinearBorderEdge')}(');
+    if (size != 1 ) {
+      s.write('size: $size');
+    }
+    if (alignment != 0 ) {
+      final String comma = size != 1 ? ', ' : '';
+      s.write('${comma}alignment: $alignment');
+    }
+    s.write(')');
+    return s.toString();
+  }
+}
+
+/// An [OutlinedBorder] like [BoxBorder] that allows one to define a rectangular (box) border
+/// in terms of zero to four [LinearBorderEdge]s, each of which is rendered as a single line.
+///
+/// The color and width of each line are defined by [side]. When [LinearBorder] is used
+/// with a class whose border sides and shape are defined by a [ButtonStyle], then a non-null
+/// [ButtonStyle.side] will override the one specified here.
```

Comment:
I don't really understand the last sentence there, can you elaborate it? (Maybe with an example?)

Reply:
using `copyWith`? Or...? I guess I don't understand what "override" means here.

Reply:
Yes, override as in reset with [OutlinedBorder.copyWith](https://api.flutter.dev/flutter/painting/OutlinedBorder/copyWith.html).